faiss/gpu/StandardGpuResources.cpp

```diff
@@ -365,8 +365,8 @@ void StandardGpuResourcesImpl::initializeForDevice(int device) {
 
     // Our code is pre-built with and expects warpSize == 32, validate that
     FAISS_ASSERT_FMT(
-            prop.warpSize == 32,
-            "Device id %d does not have expected warpSize of 32",
+            prop.warpSize == 32 || prop.warpSize == 64,
```

Comment:
Is this ROCm specific? If so, can we allow 64 only for ROCm?

Reply:
We have both wavefront 32 (E.g. navi) and 64 (E.g. MI250) devices. So this offers support for both.

Reply:
It sounds like Nvidia is 32 only and ROCm is 32 or 64. Should we lock it accordingly in code?

Reply:
If that is desired, I could rework that assert using a ROCm flag to only allow a warpSize of 64 (and 32) on ROCm devices. It shouldn't be an issue at all!

Reply:
Yeah, I think I would do that.